src/main/java/org/example/Game.java

```diff
@@ -1,181 +1,262 @@
-            }
-            System.out.println("");
-            String result = String.format("Hurray \uD83E\uDD73, %s you nailed it \uD83E\uDD29 !: ", player.getName());
-            System.out.println(result);
+  private void setPlayerName() {
+    String name;
 
-        } else {
-            for (int i = 0; i < 25; i++) {
-                System.out.print("\uD83E\uDD16 ");
-            }
-            System.out.println("");
-            String result = "Cyborg win ! Bring on your defense game \uD83D\uDCAA next time !";
-            System.out.println(result);
+    do {
+      println("Enter your name ✏\uFE0F");
+      name = this.sc.nextLine();
 
-        }
+    } while (name.isEmpty() && println("Sry, Empty name not allowed \uD83E\uDD25 !"));
+    player.setName(name);
+  }
 
-    }
+  private void setPlayerSymbol() {
+    do {
+      String result =
+          String.format("%s choose your playerSymbol from { X, O }: ", player.getName());
+      System.out.println(result);
+      player.symbol = this.sc.nextLine().toUpperCase().charAt(0);
+      if (player.symbol == 'Q') {
+        println("Are u leaving me in th middle ?? ohk then \uD83D\uDC4B \uD83D\uDC4B");
+        exit(0);
+      }
+
+    } while (!(player.symbol == X_SYMBOL || player.symbol == O_SYMBOL)
+        && println("Sry, Cannot enter anything except X_SYMBOL or O_SYMBOL \uD83E\uDD25 !"));
+  }
 
-    void initialStartup() {
+  protected void initialStartup() {
 
-        manual();
-        boardobj.initializeBoard();
-        boardobj.showBoard();
-        player.setName();
-        player.setSymbol();
-        cyborg.setSymbol( (player.getSymbol() == 'X')?'O':'X');
-        startPlay();
+    manual();
+    setPlayerName();
+    postStartup();
   }
 
-    private  void startPlay()  {
+  private void postStartup() {
+    setPlayerSymbol();
+    cyborg.setSymbol((player.getSymbol() == X_SYMBOL) ? O_SYMBOL : X_SYMBOL);
+    boardobj.initializeBoard();
+    boardobj.showBoard();
+    startPlay();
+  }
 
-        if (player.getSymbol() == 'X') {
-            while (true) {
-                playerPlay();
-                computerPlay();
-            }
+  private void startPlay() {
 
-        } else {
-            while (true) {
-                computerPlay();
-                playerPlay();
-            }
-        }
+    if (player.getSymbol() == X_SYMBOL) {
+      while (true) {
+        playerPlay();
+        computerPlay();
+      }
 
+    } else {
+      while (true) {
+        computerPlay();
+        playerPlay();
+      }
     }
+  }
 
+  private int getPositionFromUser() {
+    int position = -1;
 
+    do {
+      try {
 
-    private  void computerPlay() {
+        String result =
+            String.format(
+                "%s enter empty location for the next move from 1 to 9 : ", player.getName());
+        System.out.println(result);
+        position = this.sc.nextInt();
+        if (!(position >= 1 && position <= 9)) {
+          position = -1;
```

Comment:
You are checking positions from 1 to 9. What would you so for nXn? Here you can use your BOARD_SIZE variable to make your code extensible.

Reply:
done